packages/rs-sdk/src/platform/transition/fungible_tokens/set_price.rs

```diff
@@ -0,0 +1,174 @@
+use dpp::version::PlatformVersion;
+
+/// A builder to configure and broadcast token change direct purchase price transitions
+pub struct TokenChangeDirectPurchasePriceTransitionBuilder<'a> {
+    data_contract: &'a DataContract,
+    token_position: TokenContractPosition,
+    actor_id: Identifier,
+    token_pricing_schedule: Option<TokenPricingSchedule>,
+    public_note: Option<String>,
+    settings: Option<PutSettings>,
+    user_fee_increase: Option<UserFeeIncrease>,
+    using_group_info: Option<GroupStateTransitionInfoStatus>,
+}
+
+impl<'a> TokenChangeDirectPurchasePriceTransitionBuilder<'a> {
+    /// Start building a change direct purchase price tokens request for the provided DataContract.
+    ///
+    /// # Arguments
+    ///
+    /// * `data_contract` - A reference to the data contract
+    /// * `token_position` - The position of the token in the contract
+    /// * `issuer_id` - The identifier of the issuer
+    /// * `amount` - The amount of tokens to change direct purchase price
+    ///
+    /// # Returns
+    ///
+    /// * `Self` - The new builder instance
+    pub fn new(
+        data_contract: &'a DataContract,
+        token_position: TokenContractPosition,
+        issuer_id: Identifier,
+        token_pricing_schedule: Option<TokenPricingSchedule>,
+    ) -> Self {
+        // TODO: Validate token position
+
+        Self {
+            data_contract,
+            token_position,
+            actor_id: issuer_id,
+            token_pricing_schedule,
+            public_note: None,
+            settings: None,
+            user_fee_increase: None,
+            using_group_info: None,
+        }
+    }
+
+    /// Adds a public note to the token change direct purchase price transition
+    ///
+    /// # Arguments
+    ///
+    /// * `note` - The public note to add
+    ///
+    /// # Returns
+    ///
+    /// * `Self` - The updated builder
+    pub fn with_public_note(mut self, note: String) -> Self {
+        self.public_note = Some(note);
+        self
+    }
+
+    /// Adds a user fee increase to the token change direct purchase price transition
+    ///
+    /// # Arguments
+    ///
+    /// * `user_fee_increase` - The user fee increase to add
+    ///
+    /// # Returns
+    ///
+    /// * `Self` - The updated builder
+    pub fn with_user_fee_increase(mut self, user_fee_increase: UserFeeIncrease) -> Self {
+        self.user_fee_increase = Some(user_fee_increase);
+        self
+    }
+
+    /// Adds group information to the token change direct purchase price transition
+    ///
+    /// # Arguments
+    ///
+    /// * `group_info` - The group information to add
+    ///
+    /// # Returns
+    ///
+    /// * `Self` - The updated builder
+    pub fn with_using_group_info(mut self, group_info: GroupStateTransitionInfoStatus) -> Self {
+        self.using_group_info = Some(group_info);
+
+        // TODO: Simplify group actions automatically find position if group action is required
```

Comment:
todo :)

Reply:
all token transactions have these todos, they were originally written by Ivan. We'll clean them up at a later time.